moped-editor/src/views/projects/projectView/ProjectComponents/utils.js

```diff
@@ -36,6 +38,37 @@ export const getIntersectionLabel = (point, lines) => {
   return uniqueStreets.join(" / ");
 };
 
+const useStyles = makeStyles((theme) => ({
+  iconContainer: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    marginRight: theme.spacing(1),
+  },
+}));
+
+/**
+ * Renders an option with icon based on the type of geometry (if it exists) and component type label
+ * @param {Object} option - Autocomplete option object with label, value, and data about component type
+ * @return {JSX.Element}
+ */
+export const ComponentOptionWithIcon = ({ option }) => {
+  const classes = useStyles();
+  const { data: { line_representation = null } = {} } = option;
```

Comment:
It was hard to resist destructuring with default values here but let me know if this syntax is getting too clever and hard to read. Happy to update!

Reply:
Compared to maybe this?

```javascript
const line_representation = option?.data?.line_representation ?? null
```

I prefer the destructure! 

TIL that syntax for default assignment. I didn't not know you could do that, even though it's [right there](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#syntax) in the third example.